src/main.rs

```diff
@@ -153,48 +141,58 @@ fn search_snare_conf() -> Option<PathBuf> {
 }
 
 /// If the config specified a 'user' then switch to that and update $HOME and $USER appropriately.
+/// This function *must* not be called after daemonisation.
 fn change_user(conf: &Config) {
     match conf.user {
         Some(ref user) => match get_user_by_name(&user) {
             Some(u) => {
                 let gid = Gid::from_raw(u.primary_group_id());
                 if let Err(e) = setresgid(gid, gid, gid) {
-                    fatal_err(&format!("Can't switch to group '{}'", user), e);
+                    fatal_err(false, &format!("Can't switch to group '{}'", user), e);
                 }
                 let uid = Uid::from_raw(u.uid());
                 if let Err(e) = setresuid(uid, uid, uid) {
-                    fatal_err(&format!("Can't switch to user '{}'", user), e);
+                    fatal_err(false, &format!("Can't switch to user '{}'", user), e);
                 }
                 env::set_var("HOME", u.home_dir());
                 env::set_var("USER", user);
             }
-            None => fatal(&format!("Unknown user '{}'", user)),
+            None => fatal(false, &format!("Unknown user '{}'", user)),
         },
         None => {
             if Uid::current().is_root() {
-                fatal("The 'user' option must be set if snare is run as root");
+                fatal(
+                    false,
+                    "The 'user' option must be set if snare is run as root",
+                );
             }
         }
     }
 }
 
-/// Exit with a fatal error. This function should only be called before the [`Snare`](::Snare)
-/// struct is created.
-fn fatal(msg: &str) -> ! {
-    if msg.ends_with('.') {
-        eprintln!("{}", msg);
+/// Exit with a fatal error.
+fn fatal(daemonised: bool, msg: &str) -> ! {
```

Comment:
I just looked to see if there's a way to know if we are daemonisaed so that we don't have to pass this flag around. That's a can of worms!

The best option I found (for us) was to check if stderr is open:
https://cboard.cprogramming.com/networking-device-communication/93683-checking-if-file-descriptor-valid.html#post678998

But it seems hardly worth the effort.

Reply:
Indeed, I looked into that, and came to the same conclusion.